Automattic/newspack-ios
# Story Folder Sorting

Refs #59 

This PR makes the StoryFolder list sortable.  The UI is very much placeholder, but let's us illustrate the functionality. Things of note:
- There is a new NSUserDefaults extension that adds a `shared` singleton and should be used in favor of the normal `standard`. The `shared` singleton will return `standard` during normal operation, but for Unit tests it returns a custom instance.  This allows unit tests to not pollute defaults.
- A SortRulesBook is introduced to wrangle sort rules, serializing them to UserDefaults, and acts as a factory for NSSortDescriptors.  The SortRuleBook is designed to be reusable by different Stores simply by initializing with the values particular to a Store.

To test:
Make sure all unit tests pass. 

- View the list of story folders. 
- Create a bunch folders if necessary.
- Use the two difference sort controls.  Ensure you can sort by name and date and by ascending and descending order.
- Tap back to the menu then back to the list of stories.  Ensure the sort order was remembered and restored.
- While sorting by name, tap into a story cell to rename that story.  Confirm that the rows are correctly reordered according to the new name of the story.

@jleandroperez Game for another?
